cmd/charm/charmcmd/cmd.go

```diff
@@ -201,3 +202,42 @@ func validateAuthFlag(flagval string) (string, string, error) {
 func ussoTokenPath() string {
 	return osenv.JujuXDGDataHomePath("store-usso-token")
 }
+
+// errorMessage translates err into a user understandable error message,
+// and outputs the message ctxt.Stderr. If a message is output then
+// cmd.ErrSilent is returned, othewise err is returned unchanged.
+func errorMessage(ctxt *cmd.Context, err error) error {
+	if err == nil {
+		return err
+	}
+	cause := errgo.Cause(err)
+	switch {
+	case httpbakery.IsInteractionError(cause):
+		return interactionErrorMessage(ctxt, cause.(*httpbakery.InteractionError))
+	}
+	return err
+}
+
+// interactionErrorMessage translates err into a user understandable error message,
+// and outputs the message ctxt.Stderr.
+func interactionErrorMessage(ctxt *cmd.Context, err *httpbakery.InteractionError) error {
```

Comment:
Since this always returns ErrSilent, I wonder if it might be nicer to call
it something like:

    func showInteractionErrorMessage(ctxt *cmd.Context, err *httpbakery.InteractionError)

and return ErrSilent in the caller.

Alternatively, you could actually translate the error message:

     func translateInteractionError(err *httpbakery.InteractionError) error

and avoid printing directly to stderr entirely.
I think I prefer that - is there a particular reason to avoid the usual error printing path?